foundationdb-profiling/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+//! reaching the "chunk number"
+//!
+//! Noted: It could have more than one event in the data block
+
+use crate::parsed_key::parse_key;
+use crate::raw_transaction_profiling_block::RawTransactionProfilingBlock;
+use foundationdb::{FdbBindingError, RangeOption, Transaction};
+use futures_util::{Stream, TryStreamExt};
+use std::pin::pin;
+
+mod errors;
+mod parse;
+mod parsed_key;
+mod raw_transaction_profiling_block;
+mod scanner;
+
+/// Defined by the following [pattern](https://github.com/apple/foundationdb/blob/main/contrib/transaction_profiling_analyzer/transaction_profiling_analyzer.py#L421)
+pub const PROFILE_PREFIX: &[u8; 32] = b"\xff\x02/fdbClientInfo/client_latency/";
+
+/// Retrieves raw profiling datablocks stored in the database within the specified read version range.
+///
+/// This function fetches raw events from the FoundationDB range key-value store. The keys are
+/// processed to parse the profiling event data. Events are grouped into chunks and consolidated
+/// into complete [RawTransactionProfilingBlock] instances. The resulting stream yields these complete profiling event
+/// data blocks. Events are referenced using a key structure which includes the transaction ID,
+/// version stamp, and a chunk identifier.
+///
+/// # Arguments
+///
+/// * `trx` - A reference to the FoundationDB `Transaction` used for fetching data.
+/// * `start_read_version` - An optional lower bound read version. If provided, only events
+///                          occurring at or after this version are fetched.
+/// * `end_read_version` - An optional upper bound read version. If provided, only events
+///                        occurring before this version are fetched.
+///
+/// # Returns
+///
+/// Returns an asynchronous stream (`impl Stream`) where each item of the stream is either:
+/// * A `Result<DataBlock, FdbBindingError>`
+///   - `DataBlock`: A complete profiling event data block.
+///   - `FdbBindingError`: An error that occurred during data retrieval or processing.
+///
+/// # Errors
+///
+/// This function will return an error in the following cases:
+/// * If the transaction (provided by `trx`) encounters issues fetching ranges.
+/// * If an error occurs during key or value parsing.
+///
+/// # Notes
+///
+/// Profiling keys follow a specific binary layout defined by the FoundationDB profiling data space:
+/// * `\xff\x02/fdbClientInfo/client_latency/` acts as the prefix for profiling events.
+/// * Events are referred to by tuple: (VersionStamp, TransactionId).
+/// * Events contain chunks, defined by a chunk number and total chunks.
+///
+/// The function ensures fully constructed events are yielded only once all their chunks are retrieved.
+///
+/// # Example
+///
+/// ```ignore
+/// use foundationdb::{Database, Transaction};
+/// use futures_util::StreamExt;
+///
+/// async fn example_usage(trx: &Transaction) {
+///     let mut stream = get_raw_events(trx, Some(1), Some(10));
```

Comment:
suggestion: The example is not compiling(`get_raw_events` is now `get_raw_datablocks`), so we should either:

* Remove the ignore tags to fail on CI for such types of errors,
* Remove the example